boards/arm/cy8cproto_062_4343w/board.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021 Cypress Semiconductor Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <cybsp.h>
+#include <sys/__assert.h>
+#include <init.h>
+
+
+
+static int board_init(const struct device *dev)
+{
+	cy_rslt_t result;
+
+	/* Initializes the board support package */
+	result = cybsp_init();
```

Comment:
why is "board support package" needed in Zephyr?

Reply:
Right, in Zephyr we do not have "BSP"s, we just have "boards".

Reply:
@gmarull and @carlescufi,

The cybsp_* APIs are APIs from our SOC SDK. They are used to act on a board (represented by a BSP) to initialize or control peripherals. Rather than reimplement how we initialize our board in different environments, we just reuse the SDK.

Does this make sense and address your concerns?

Reply:
I think this PR is missing some fundamental Zephyr concepts. There's no "BSP" in Zephyr. Boards define instead the enabled SoC peripherals or other hardware devices in Devicetree. Then, drivers _pick_ this information to setup the system accordingly. We still have some boards doing manual pinmux, something that could be seen as "BSP code", but still this will soon be deprecated because there's now a pinctrl API. So the approach taken here has to be reworked to follow this model. Note that you can still plug init code into the SoC init function, if needed.

Reply:
The board uses GPIO, UART  from Devicetree configuration. Another peripherals support (ADC, SPI, I2C, etc) will be done in future PRs.

"bsp" was reused from existing Asset (https://github.com/Infineon/TARGET_CY8CPROTO-062-4343W) to do board configuration of system resources (PLL, FLL, WCO, CM4/CM0p clock configuration,  SWD configuration, power, etc).

so we can do following:
1. move 'bsp' from board zephyr\boards\arm\cy8cproto_062_4343w\ to hal_infenion\bsp\TARGET_CY8CPROTO-062-4343W
2. call cybsp_init() from zephyr\soc\arm\infineon_cat1\psoc6\soc.c.

Reply:
No, you don't need any BSP code. You need drivers that configure your peripherals based on DT configuration. Sometimes, you'll need some init code part of soc.c, but not a BSP hidden within soc.c.